File: datavar.dart
void main() {
  // Integer
  int myInt = 42;
  print('Integer: $myInt');

  // Float
  double myFloat = 3.14;
  print('Float: $myFloat');

  // Double
  double myDouble = 3.14159;
  print('Double: $myDouble');

  // Character (Dart uses string for single characters)
  String myChar = 'A';
  print('Char: $myChar');

  // String
  String myString = 'Hello, Dart!';
  print('String: $myString');

  // Boolean
  bool myBoolean = true;
  print('Boolean: $myBoolean');

  // Array (List)
  List<int> myArray = [1, 2, 3, 4, 5];
  print('Array: $myArray');

  // Map (Dictionary)
  Map<String, int> myMap = {'one': 1, 'two': 2, 'three': 3};
  print('Map: $myMap');

  // Tuple (using a list to represent a tuple)
  List<dynamic> myTuple = [42, 'Dart', true];
  print('Tuple: $myTuple');

  // Function
  int myFunction(int x) => x * 2;
  print('Function (2 * 3): ${myFunction(3)}');

  // Class (Struct equivalent)
  class Person {
    String name;
    int age;

    Person(this.name, this.age);
  }

  Person myPerson = Person('Alice', 25);
  print('Class: Name = ${myPerson.name}, Age = ${myPerson.age}');

  // Null
  String? myNull = null;
  print('Null: $myNull');
}

File: datavar.ts
// Integer
let myInt: number = 42;
console.log("Integer:", myInt);

// Float
let myFloat: number = 3.14;
console.log("Float:", myFloat);

// Double (TypeScript uses number for both float and double)
let myDouble: number = 3.14159;
console.log("Double:", myDouble);

// Char (TypeScript uses string for characters)
let myChar: string = 'A';
console.log("Char:", myChar);

// String
let myString: string = "Hello, TypeScript!";
console.log("String:", myString);

// Boolean
let myBoolean: boolean = true;
console.log("Boolean:", myBoolean);

// Array
let myArray: number[] = [1, 2, 3, 4, 5];
console.log("Array:", myArray);

// Tuple
let myTuple: [number, string, boolean] = [42, "TypeScript", true];
console.log("Tuple:", myTuple);

// Function
let myFunction = (x: number): number => x * 2;
console.log("Function (2 * 3):", myFunction(3));

// Interface (Struct equivalent)
interface Person {
    name: string;
    age: number;
}
let myPerson: Person = { name: "Alice", age: 25 };
console.log("Struct: Name =", myPerson.name, "Age =", myPerson.age);

// Null
let myNull: null = null;
console.log("Null:", myNull);
